### PR TITLE
utilizes json for transferring data

### PR DESCRIPTION
**Pull request by connorbrett**
step 3 from walkthrough

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -18,6 +18,7 @@
 
 function getMessage() {
   fetch('/data').then(response => response.text()).then(msg => {
+    console.log(msg);
```
Comment (fenghaolw) [inline portfolio/src/main/webapp/script.js:21]: Shouldn't you use `response.json()` now? The server is returning a json.

Comment (fenghaolw) [inline portfolio/src/main/webapp/script.js:21]: Remove `console.log`? It should only be used for debugging.

Comment (connorbrett) [reply portfolio/src/main/webapp/script.js:21]: not sure, I was just following what they said in the walkthrough

Comment (fenghaolw) [reply portfolio/src/main/webapp/script.js]: The walkthrough mentioned parsing json: https://github.com/connorbrett/step/blob/master/walkthroughs/week-3-server/step-3-json-walkthrough.md#parsing-json

Comment (connorbrett) [reply portfolio/src/main/webapp/script.js]: ok that makes more sense, misread it the first time i went through it, fixed now

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -17,7 +17,7 @@
 */
 
 function getMessage() {
-  fetch('/data').then(response => response.text()).then(msg => {
+  fetch('/data').then(response => response.json()).then(msg => {
```
Comment (fenghaolw) [inline portfolio/src/main/webapp/script.js:20]: So now `msg` is a JSON array and you are implicitly depending on its `toString()` method to print it out.

You could make it better by constructing HTML elements for each of the message, but I won't block this PR for that.

**Pull request merged**